pkg/proto/blobstore/blobstore.proto

```diff
@@ -42,6 +42,10 @@ message BlobAccessConfiguration {
         // Fan out requests across multiple storage backends to spread
         // out load.
         ShardingBlobAccessConfiguration sharding = 9;
+
+        // Delegate requests to an underlying backend but forward
+        // incoming client metadata (including credentials).
+        GRPCMetadataForwardingBlobAccessConfiguration grpc_metadata_forwarding = 10;
```

Comment:
Do we expect any configuration options to appear for this adapter? If not, we could consider not having a `GRPCMetadataForwardingBlobAccessConfiguration` at all. Instead, we could embed a `BlobAccessConfiguration`. This has the advantage of preventing the need for an unnecessary `remote {}` block in the configuration.